File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int index=-1;
        int len=haystack.length()-needle.length()+1;
        outter:for(int i=0;i<len;i++){
            inner:for(int j=0;j<needle.length();j++){
                if(haystack.charAt(i+j)==needle.charAt(j)){
                    index=i;
                    if(j==needle.length()-1){
                        break outter;
                    }
                }else{
                    index=-1;
                    break inner;
                }
            }
        }
        return index;
    }
}

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        l=len(haystack)-len(needle)+1
        for i in range(0,l):
            for j in range(len(needle)):
                if(haystack[i+j]!=needle[j]):
                    break 
                if(haystack[i+j]==needle[j] and j==len(needle)-1):
                    return i
        return -1
                
        